[processing] Remove duplicate fill holes algorithm

nyalldawson opened the pull request:
This PR consolidates the duplicate fill holes algorithms into one algorithm, by:

- moving guts of the algorithm to QgsGeometry (with tests)
- make algorithm maintain z/m values
- make algorithm work with curved geometries
- add unit tests for algorithm

The extra Fill_holes algorithm can now safely be removed. I can't work out if processing scripts can be deprecated, but in any case when we have migration path worked out this can then be removed.

nyalldawson committed:
Add method to QgsGeometry to remove interior rings from a polygon

With an optional area threshold

nyalldawson committed:
[processing] Consolidate 'delete holes' algorithms

Copy min area parameter from 'Fill holes' algorithm to 'delete
holes' algorithm.

Also:

- make algorithm maintain z/m values
- make algorithm work with curved geometries
- add unit tests

nirvn commented on python/core/geometry/qgscurvepolygon.sip:
```diff
@@ -57,6 +57,14 @@ class QgsCurvePolygon: public QgsSurface
+     * area will be removed.
+     * @note added in QGIS 3.0
+     */
+    void removeInteriorRings( double minimumRingArea = -1 );
```
@nyalldawson , I'm wondering whether it'd be useful to have a maximumRingArea parameter too. Say you want to simplify a feature to display nicely at a given zoomed-out level, you might want to remove rings that are below a maximumArea threshold.

nyalldawson replied:
That's what the minimumRingArea parameter does - it specifies the minimum allowable ring size

nyalldawson replied:
Fixed

nyalldawson committed:
Rename minimum area paremeter, fix docs for QgsCurvePolygon::removeInteriorRing

volaya commented:
@nyalldawson There is no option to show the script as deprecated, but i think it can safely be removed. In any case, we have API breaks like that in other places in Processing, for 3.0, so no need to worry. This is a minor issue, since the algorithm is not likely to be used in many models anyway

volaya commented:
+1 from me to merge

nyalldawson committed:
[FEATURE][processing] Remove duplicate fill holes algorithm

The Fill_holes script is duplicated by the DeleteHoles algorithm,
and the DeleteHoles algorithm has many advantages:
- it's a full algorithm rather than a script
- uses qgis geometry classes and maintains curves/z/m
- has unit tests
- doesn't rely on non-standard external library (shapely)

(marked as feature for inclusion in release notes)

nyalldawson committed:
[FEATURE][processing] Remove duplicate create points along lines script

This functionality is duplicated by the PointsAlongGeometry algorithm

Keep the algorithm version, since it has unit tests and supports things
like progress reports.

(marked as feature for inclusion in release notes)

nyalldawson commented:
@volaya - ok, merged. I've also removed a duplicate create points along lines script.

nirvn commented:
Nice one.
